spring-web/src/main/java/org/springframework/web/context/support/WebApplicationContextUtils.java

```diff
@@ -125,6 +136,63 @@ public static WebApplicationContext getWebApplicationContext(ServletContext sc,
 		return (WebApplicationContext) attr;
 	}
 
+	/**
+	 * Registers a custom {@link WebApplicationContext} for this web app.
+	 * @param sc ServletContext to find the web application context for
+	 * @param attrName the name of the ServletContext attribute to use
+	 * @param context the desired WebApplicationContext to store
+	 * @since 4.2
+	 */
+	public static void registerWebApplicationContext(ServletContext sc, String attrName, WebApplicationContext context) {
+		Assert.notNull(sc, "ServletContext must not be null");
+		Assert.notNull(attrName, "attrName must not be null");
+		Assert.notNull(context, "WebApplicationContext must not be null");
+
+		sc.setAttribute(attrName, context);
+
+		@SuppressWarnings("unchecked")
+		LinkedHashSet<String> attrNames = (LinkedHashSet<String>) sc.getAttribute(APPLICATION_CONTEXT_ATTRIBUTES_ATTRIBUTE);
+		if(attrNames == null) {
```

Comment:
Thanks! This should be fixed now. Is there an IntelliJ code formatter available?